internal/jimm/role.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2024 Canonical.
+package jimm
+
+import (
+	"context"
+
+	"github.com/canonical/jimm/v3/internal/common/pagination"
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/openfga"
+)
+
+// AddRole creates a role within JIMMs DB for reference by OpenFGA.
+func (j *JIMM) AddRole(ctx context.Context, user *openfga.User, name string) (*dbmodel.RoleEntry, error) {
+	return &dbmodel.RoleEntry{}, nil
+}
```

Comment:
I guess you still need to hook all of these up to the DB methods. Will that be for this PR or a follow-up?

Reply:
yes, the jimm implementation is for a follow-up.
So this 300lines skeleton pr is not distracting from the actual implementation.